alacritty/src/config/monitor.rs

```diff
@@ -15,18 +14,34 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
 #[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
 const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
 
-pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
-    // Canonicalize all paths, filtering out the ones that do not exist.
-    paths = paths
-        .drain(..)
-        .filter_map(|path| match fs::canonicalize(&path) {
-            Ok(path) => Some(path),
-            Err(err) => {
-                error!("Unable to canonicalize config path {:?}: {}", path, err);
-                None
-            },
-        })
-        .collect();
+pub fn watch(paths: Vec<PathBuf>, event_proxy: EventProxy) {
+    let paths = {
+        let mut paths_tmp = Vec::new();
```

Comment:
The unfortunate naming of this variable indicates that this is probably not the greatest way to solve this problem. I don't think it should be necessary to create a separate vector.

Reply:
My Rust knowledge is probably at fault here, but the alternatives I see are:
- using iterators, which runs into the problem I was describing
- modifying the vector in place, which sounds a lot more cumbersome and inefficient than creating a new one

Reply:
Just to clarify: I'll try and play around with a few options myself when I got the time. I just need a little time until I can get around to it.